hcl2template/types.variables.go

```diff
@@ -75,44 +75,40 @@ func (variables Variables) Values() map[string]cty.Value {
 	return res
 }
 
-// decodeConfig decodes a "variables" section the way packer 1 used to
-func (variables *Variables) decodeConfigMap(block *hcl.Block, ectx *hcl.EvalContext) hcl.Diagnostics {
+// decodeVariable decodes a variable key and value into Variables
+func (variables *Variables) decodeVariable(key string, attr *hcl.Attribute, ectx *hcl.EvalContext) hcl.Diagnostics {
+	var diags hcl.Diagnostics
+
 	if (*variables) == nil {
 		(*variables) = Variables{}
 	}
-	attrs, diags := block.Body.JustAttributes()
 
-	if diags.HasErrors() {
+	if _, found := (*variables)[key]; found {
+		diags = append(diags, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Duplicate variable",
+			Detail:   "Duplicate " + key + " variable definition found.",
+			Subject:  attr.NameRange.Ptr(),
+		})
 		return diags
 	}
 
-	for key, attr := range attrs {
-		if _, found := (*variables)[key]; found {
-			diags = append(diags, &hcl.Diagnostic{
-				Severity: hcl.DiagError,
-				Summary:  "Duplicate variable",
-				Detail:   "Duplicate " + key + " variable definition found.",
-				Subject:  attr.NameRange.Ptr(),
-				Context:  block.DefRange.Ptr(),
```

Comment:
I had to remove this Context from the diagnose. Is it something super important?

Reply:
The context in a diagnostic allows HCL to tell where the duplicate variable definition is in the config file; it is nice for the user experience but will not make Packer crash or anything.